core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -376,6 +397,10 @@ public void transportTerminated() {
   public static final Metadata.Key<Long> TIMEOUT_KEY =
       Metadata.Key.of(HttpUtil.TIMEOUT, new TimeoutMarshaller());
 
+  // TODO(carl-mastrangelo): move this to internal
+  public static final Metadata.Key<String> MESSAGE_ENCODING_KEY =
+      Metadata.Key.of(HttpUtil.MESSAGE_ENCODING, Metadata.ASCII_STRING_MARSHALLER);
```

Comment:
Ah I see you already had a TODO for this ... so I gave you the appropriate place to put it :)